src/main/java/org/wso2/carbon/module/swiftiso20022/utils/ConnectorUtils.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.module.swiftiso20022.utils;
+
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.Constants;
+import org.apache.axis2.context.MessageContext;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.commons.json.JsonUtil;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.transport.passthru.PassThroughConstants;
+import org.apache.synapse.transport.passthru.util.RelayUtils;
+import org.wso2.carbon.connector.core.ConnectException;
+import org.wso2.carbon.module.swiftiso20022.constants.ConnectorConstants;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * Class to hold the utility methods.
+ */
+public class ConnectorUtils {
+    private static final Log log = LogFactory.getLog(ConnectorUtils.class);
+
+    /**
+     * Build Message and extract payload.
+     *
+     * @param axis2MC message context
+     * @return optional json message
+     * @throws ConnectException thrown if unable to build
+     */
+    public static Optional<String> buildMessagePayloadFromMessageContext(MessageContext axis2MC)
+            throws ConnectException {
+        String requestPayload = null;
+        boolean isMessageContextBuilt = isMessageContextBuilt(axis2MC);
```

Comment:
Without this variable assignment, shall we directly call this method inside the if check?

Reply:
Fixed in